### PR TITLE
 [unitaryhack] Add decomposition for qml.CSWAP #1306 

### PR DESCRIPTION
**Pull request by MaldoAlberto**
I am Alberto Maldonado and I added the definition of CSWAP components from three Toffoli gates.

### CODE REVIEW

**Comment by josh146** [comment]
Thanks @MaldoAlberto for this contribution!

I've noticed that a few of the checks are currently failing:

- Changelog: don't forget to add an entry to the changelog (`.github/CHANGELOG.md`) describing this change, as well as linking back to this PR. You can also add your name to the contributors section 🙂 

- Formatting: If you haven't already, install the `black` code formatter: `pip install black`. Then, before you commit changes, make sure to format the source code and tests: `black -l 100 pennylane tests`

- Tests: finally, it looks like the new test is not passing, resulting in the following error message:

  ```python
   Traceback (most recent call last):
   File "/home/runner/work/pennylane/pennylane/tests/ops/test_qubit_ops.py", line 744, in test_CSWAP_decomposition
      [[1, 0, 0, 0, 0, 0, 0, 0]
  TypeError: list indices must be integers or slices, not tuple
  ```

Let us know when you have addressed these, and someone from the team will be on hand to provide a review 🙂 In addition, feel free to ask us any questions!

**Comment by MaldoAlberto** [comment]
Hi @josh146  I have already made the pertinent changes but I got this in files that I did not modify
Screenshot from 2021-05-20 11-34-57

**Comment by josh146** [comment]
Hi @MaldoAlberto, that's strange 🤔 Could you try updating your version of black,
```
pip install black --upgrade
```
and trying again?

**Comment by MaldoAlberto** [comment]
@josh146 now works 
Screenshot from 2021-05-20 13-24-59

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -118,7 +119,9 @@
   + (0.25) [Y5 X4 Y0]
   >>> mapping
   {0: (0, 1), 1: (0, 2), 2: (1, 0), 3: (1, 2), 4: (2, 0), 5: (2, 1)}
+
```
Comment (josh146) [inline .github/CHANGELOG.md:122]:
```suggestion
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Release 0.16.0-dev (development release)
 
+
```
Comment (josh146) [inline .github/CHANGELOG.md:3]:
```suggestion
```

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -119,6 +119,7 @@
   >>> mapping
   {0: (0, 1), 1: (0, 2), 2: (1, 0), 3: (1, 2), 4: (2, 0), 5: (2, 1)}
   ```
+  Add decomposition for qml.CSWAP
```
Comment (josh146) [inline .github/CHANGELOG.md:122]: @MaldoAlberto it looks like this was added inside an existing changelog entry 🙂 

Could you move this down to it's own bullet point in the "Improvements" section? Thanks! Don't forget to also provide a link back to this PR, and to include your name in the 'Contributors" section :)

File: pennylane/ops/qubit.py [hunk 4]
```diff
@@ -634,6 +634,15 @@ class CSWAP(Operation):
...
         return cls.matrix
 
+    @staticmethod
+    def decomposition(wires):
+        decomp_ops = [
+            qml.Toffoli(wires=[wires[0], wires[2], wires[1]]),
+            qml.Toffoli(wires=[wires[0], wires[1], wires[2]]),
+            qml.Toffoli(wires=[wires[0], wires[2], wires[1]]),
+        ]
+        return decomp_ops
```
Comment (josh146) [inline pennylane/ops/qubit.py:644]: Nice!

**Review by josh146 — CHANGES_REQUESTED** [review]
Thanks @MaldoAlberto for this great contribution. It is looking almost ready for merging, just a slight update needed for the changelog first!

File: .github/CHANGELOG.md (hunk 3 above)
Comment (MaldoAlberto) [reply .github/CHANGELOG.md]: @josh146  I did it   :+1:

File: pennylane/ops/qubit.py (hunk 4 above)
Comment (MaldoAlberto) [reply pennylane/ops/qubit.py:644]: I hope to contribute with more improvements :D

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -174,6 +173,8 @@ random_mat2 = rng.standard_normal(3, requires_grad=False)
 
 <h3>Improvements</h3>
 
+*  Define decomposition for qml.CSWAP [(#1306)](https://github.com/PennyLaneAI/pennylane/issues/1306)
```
Comment (josh146) [inline .github/CHANGELOG.md:176]:
```suggestion
*  A decomposition has been added for the `qml.CSWAP` operation.
  [(#1306)](https://github.com/PennyLaneAI/pennylane/issues/1306)
```

File: pennylane/ops/qubit.py (hunk 4 above)
Comment (josh146) [reply pennylane/ops/qubit.py:644]: Once this is merged in, feel free to get started on any other of our UnitaryHACK labeled issues!

**Review by josh146 — APPROVED** [review]
Thanks again @MaldoAlberto, this is now ready for merging 🎉

**Pull request merged**